java/ql/src/experimental/semmle/code/java/frameworks/Jndi.qll

```diff
@@ -0,0 +1,11 @@
+import java
+
+/** The class `javax.naming.InitialContext`. */
+class TypeInitialContext extends Class {
+  TypeInitialContext() { this.hasQualifiedName("javax.naming", "InitialContext") }
+}
+
+/** The class `javax.naming.CompositeName`. */
+class TypeCompositeName extends Class {
+  TypeCompositeName() { this.hasQualifiedName("javax.naming", "CompositeName") }
```

Comment:
Should account for `javax.naming.Name` interface instead? Not sure if `CompositeName` is the only one that can hold a JNDI inj payload or if it can be hold in a `Name` object at all (maybe only String params to lookup are valid for injection purposes)

Reply:
`Name` interface has 3 implementations in JDK: `CompositeName`, `CompoundName` and `LdapName`. `CompositeName` and `CompoundName` can hold JNDI payload:

```java
ctx.lookup(new CompositeName("rmi:\\/\\/127.0.0.1:12345/abc"));
ctx.lookup(new CompoundName("rmi://127.0.0.1:12345/abc", new Properties()));
```

`LdapName`, most likely, can't.

I'll add support for `CompoundName` in the query, but if you think that handling all `javax.naming.Name` implementations is better, I can change it.

Reply:
If `LdapName` cant, I think its enough with adding `CompoundName`, thanks!